product-service/src/functions/get-product/handler.ts

```diff
@@ -0,0 +1,26 @@
+import 'source-map-support/register';
+
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import { ErrorResponse, formatJSONResponse } from '@libs/apiGateway';
+import { middyfy } from '@libs/lambda';
+
+import { PRODUCT_BY_ID_MAP } from 'src/products-list';
+import { HttpCode } from 'src/constants';
+
+export const getProduct: APIGatewayProxyHandler = async event => {
+    const productId = event.pathParameters?.productId;
+
+    if (productId != null) {
+        const product = PRODUCT_BY_ID_MAP.get(productId);
+
+        if (product != null) {
+            return formatJSONResponse({
+                product,
+            });
+        }
+    }
```

Comment:
u could use try catch block 

e.g. emulate service for fetching products as:

const getProducts = () => Promise.resolve(YOUR_MOCK)

and then here wrap it in tryCatch

try {
  const products = await getProducts()
} catch {}